test/model/message_list_test.dart

```diff
@@ -2244,8 +2813,16 @@ void checkInvariants(MessageListView model) {
   check(model).middleItem
     ..isGreaterOrEqual(0)
     ..isLessOrEqual(model.items.length);
-  if (model.middleItem == model.items.length) {
-    check(model.middleMessage).equals(model.messages.length);
+  if (model.middleMessage == model.messages.length) {
+    if (model.outboxMessages.isEmpty) {
+      // the bottom slice of `model.messages` is empty
+      check(model.items).length.equals(model.middleItem);
```

Comment:
nit: this check is really about validating the behavior of `middleItem`, not `items`
```suggestion
      check(model.middleItem).equals(model.items.length);
```